CMakeLists.txt

```diff
@@ -1,178 +1,152 @@
-# per-thread default stream option
-# This needs to be defined first so tests and benchmarks can inherit it.
-
-if(PER_THREAD_DEFAULT_STREAM)
-    message(STATUS "Using per-thread default stream")
-    add_compile_definitions(CUDA_API_PER_THREAD_DEFAULT_STREAM)
-endif(PER_THREAD_DEFAULT_STREAM)
-
-###################################################################################################
 # library targets
 
 add_library(rmm INTERFACE)
 add_library(rmm::rmm ALIAS rmm)
 
-target_include_directories(rmm INTERFACE
-  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-  "$<INSTALL_INTERFACE:include>"
-  )
+target_include_directories(rmm INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+                                         "$<INSTALL_INTERFACE:include>")
 
 if(CUDA_STATIC_RUNTIME)
-  message(STATUS "Enabling static linking of cudart")
+  message(STATUS "RMM: Enabling static linking of cudart")
   target_link_libraries(rmm INTERFACE CUDA::cudart_static)
 else()
   target_link_libraries(rmm INTERFACE CUDA::cudart)
 endif(CUDA_STATIC_RUNTIME)
 
 target_link_libraries(rmm INTERFACE rmm::Thrust)
 target_link_libraries(rmm INTERFACE spdlog::spdlog_header_only)
+target_compile_features(rmm INTERFACE cxx_std_14 cuda_std_14)
+target_compile_options(rmm INTERFACE $<$<COMPILE_LANG_AND_ID:CXX,GNU,Clang>:-Werror
+                                     -Wno-error=deprecated-declarations>)
 
-###################################################################################################
-# Set logging level. Must go before including gtests and benchmarks. 
+if(DISABLE_DEPRECATION_WARNING)
+  target_compile_options(rmm INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler
+                                       -Wno-deprecated-declarations>)
+  target_compile_options(rmm INTERFACE $<$<COMPILE_LANGUAGE:CXX>:-Wno-deprecated-declarations>)
+endif(DISABLE_DEPRECATION_WARNING)
 
-set(RMM_LOGGING_LEVEL "INFO" CACHE STRING "Choose the logging level.")
-# Set the possible values of build type for cmake-gui
-set_property(CACHE RMM_LOGGING_LEVEL PROPERTY STRINGS
-    "TRACE" "DEBUG" "INFO" "WARN" "ERROR" "CRITICAL" "OFF")
-message(STATUS "RMM_LOGGING_LEVEL = '${RMM_LOGGING_LEVEL}'.")
+if(PER_THREAD_DEFAULT_STREAM)
+  message(STATUS "RMM: Using per-thread default stream")
+  target_compile_definitions(rmm INTERFACE CUDA_API_PER_THREAD_DEFAULT_STREAM)
+endif(PER_THREAD_DEFAULT_STREAM)
 
-###################################################################################################
-# add gtest
+# Set logging level. Must go before including gtests and benchmarks.
+
+set(RMM_LOGGING_LEVEL
+    "INFO"
+    CACHE STRING "Choose the logging level.")
+# Set the possible values of build type for cmake-gui
+set_property(
+  CACHE RMM_LOGGING_LEVEL
+  PROPERTY STRINGS
+           "TRACE"
+           "DEBUG"
+           "INFO"
+           "WARN"
+           "ERROR"
+           "CRITICAL"
+           "OFF")
```

Comment:
I guess this is auto-formatted, so I guess it's fine, otherwise it's using too many lines for my taste ;)

Reply:
I could potentially play with this more, but in general I think the rule is that if the command will span more than 1 line horizontally it will switch to a vertical layout instead

Reply:
Was able to tweak this to get a more compact layout in situations like this while still going vertical when it makes sense to do so.